logistic_requisition/model/sale_order.py

```diff
@@ -29,6 +30,103 @@ class SaleOrder(models.Model):
                                      ondelete='restrict',
                                      copy=False)
 
+    lr_count = fields.Integer(
+        compute='_count_logistic_requisition'
+    )
+    lr_source_count = fields.Integer(
+        compute='_count_lr_source'
+    )
+    lr_purchase_count = fields.Integer(
+        compute='_count_lr_purchase'
+    )
+
+    @api.multi
+    def action_accepted(self):
+        """ On acceptation of Cost Estimate, we generate PO
+        for all line sourced by a tender
+
+        """
+        res = super(SaleOrder, self).action_accepted()
+        purch_req_so_lines = self.mapped('order_line').filtered(
+            lambda rec: (rec.sourcing_method == 'procurement'
+                         and not rec.sourced_by))
+        todo = self.env['purchase.requisition'].browse()
```

Comment:
You are right `Environment.__getitem__(self, model_name)` already call `_browse`